full/src/test/java/apoc/coll/CollFullTest.java

```diff
@@ -0,0 +1,32 @@
+package apoc.coll;
+
+import apoc.util.TestUtil;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.neo4j.test.rule.DbmsRule;
+import org.neo4j.test.rule.ImpermanentDbmsRule;
+import org.neo4j.values.storable.DurationValue;
+
+import static org.junit.Assert.assertEquals;
+
+public class CollFullTest {
```

Comment:
I would like to see more test coverage, at least the following inputs to apoc.coll.avgDuration:

- empty list - should return null
- null - should return null?
- a list of another type e.g. [1, 2, 3] - should throw some helpful error
- a list with mixed durations and numbers e.g. [duration(''P2DT4H1S'), 2] - should throw some helpful error
- big enough durations and/or long enough list to cause an overflow problem without running avg (see comment on the implementation, I think there is a limit when the sum of all of the durations hit Long.MAX seconds)

Reply:
It would maybe also make sense to document some of these edge case behaviours.

Reply:
Added